tdigest.go

```diff
@@ -335,3 +345,96 @@ func (cs *centroidSet) MergeInto(other TDigest) {
 		other.Add(c.mean, c.count)
 	}
 }
+
+func (cs *centroidSet) MarshalBinary() ([]byte, error) {
+	buf := bytes.NewBuffer(nil)
+	w := &stickyErrWriter{w: buf, err: nil}
```

Comment:
I really like that you made a type to help here. The boilerplate of `binary.Write(w, binary.LittleEndian, v)` remains; can the custom type encompass that as well, exposing a method like `WriteValue(v interface{})`?

Reply:
Done, good idea.